src/Controller/ExportController.php

```diff
@@ -137,4 +110,33 @@ private function getFont( Request $request, $lang ): ?string {
 		}
 		return $font;
 	}
+
+	/**
+	 * Error page handler, to always show the export form with any HTTP error message.
+	 *
+	 * @param Api $api
+	 * @param Exception $exception
+	 * @return Response
+	 */
+	public function error( Throwable $exception, Api $api ) {
+		$message = $exception->getMessage();
+		if ( $exception instanceof RequestException ) {
+			$exceptionResponse = $exception->getResponse();
+			if ( $exceptionResponse ) {
+				$message = Util::extractErrorMessage( $exceptionResponse, $exception->getRequest() );
+			}
+		}
+		return $this->render( 'export.html.twig', [
```

Comment:
Perhaps we could check the env and only do this in production? For 500s especially I think it's ideal that we can see the stack trace without having to check the logs. I also quite like Symfony's exception interface where it conveniently highlights the problematic lines of code.

Reply:
You don't have to check the logs, you can just click the debug toolbar and get it in a readable display. But I know what you mean! How about we just catch Http and Request exceptions? Those are more likely to be ones that the user can recover from (e.g. wrong page title or wiki lang code).

Reply:
I've updated it to just check Http and Request (the traces of which are still visible via the toolbar).